apm-lambda-extension/extension/route_handlers.go

```diff
@@ -30,18 +32,28 @@ type AgentData struct {
 }
 
 var AgentDoneSignal chan struct{}
+var mainExtensionContext context.Context
 
 // URL: http://server/
-func handleInfoRequest(apmServerUrl string) func(w http.ResponseWriter, r *http.Request) {
+func handleInfoRequest(ctx context.Context, apmServerUrl string, config *extensionConfig) func(w http.ResponseWriter, r *http.Request) {
 	return func(w http.ResponseWriter, r *http.Request) {
 
+		Log.Debug("Handling APM Server Info Request")
+		mainExtensionContext = ctx
+
 		// Init reverse proxy
 		parsedApmServerUrl, err := url.Parse(apmServerUrl)
 		if err != nil {
 			Log.Errorf("could not parse APM server URL: %v", err)
 			return
 		}
+
 		reverseProxy := httputil.NewSingleHostReverseProxy(parsedApmServerUrl)
+		reverseProxyTimeout := time.Duration(config.DataForwarderTimeoutSeconds) * time.Second
+		reverseProxy.Transport = &http.Transport{
+			ResponseHeaderTimeout: reverseProxyTimeout,
+		}
```

Comment:
As mentioned offline, please do not overwrite the whole `http.Transport` here, but rather only set the `ResponseHeaderTimeout`, or even pass it as an option above when calling `httputil.NewSingleHostReverseProxy`.